community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/SessionStateMachine.java

```diff
@@ -584,6 +587,25 @@ public boolean hasTransaction()
         return currentTransaction != null;
     }
 
+    @Override
+    public QuerySession createSession( GraphDatabaseQueryService service, PropertyContainerLocker locker )
+    {
+        InternalTransaction transaction =
+                service.beginTransaction( currentTransaction.transactionType(), currentTransaction.mode() );
+        Neo4jTransactionalContext transactionalContext =
+                new Neo4jTransactionalContext( service, transaction, txBridge.get(), locker );
+
+        return new QuerySession( transactionalContext )
+        {
+
+            @Override
+            public String toString()
+            {
+                return "bolt";
```

Comment:
This could be a bit more helpful. What about `QuerySession for Bolt`, possibly using something from `transactionalContext`?